sources/updater/updog/src/main.rs

```diff
@@ -262,6 +269,16 @@ fn update_flags() -> Result<()> {
     Ok(())
 }
 
+fn revert_update_flags() -> Result<()> {
+    let mut gpt_state = State::load().context(error::PartitionTableRead)?;
+    // This actually wipes all the priority bits in the inactive partition
+    gpt_state.cancel_upgrade();
+    // The update is still in the inactive partition so mark the inactive partition as valid
+    gpt_state.mark_inactive_valid();
```

Comment:
B in this case should still be valid because of the `mark_inactive_valid` call. You can immediately call `update-apply` again to make B boot next. Without the `mark-inactive-valid` call here, B will actually be dead.
I think the testing description in the PR description for updog testing shows this as well.

Reply:
I tried addressing this by checking in `update_revert` if the inactive partition was previously valid before actually marking as valid after clearing the flags.
See https://github.com/bottlerocket-os/bottlerocket/pull/942/commits/702cdddbc103f71e733cb71dcad10055169d2e6a